### PR TITLE
Temporal Motif Mining

### PR DESCRIPTION
**Pull request by tanmoyio**
dummy script
issue: https://github.com/graphistry/pygraphistry/issues/416
```python3
import graphistry
import pandas as pd

edges = pd.DataFrame([
    [0, 1, 2],
    [5, 2, 3],
    [10, 2, 4],
    [15, 5, 6],
    [20, 6, 7],
    [25, 6, 8],
    [30, 7, 4]], 
    columns=['time', 'src', 'dst']
)
g = graphistry.edges(edges, 'src', 'dst')
print(g.motif('time', accumulate=10, step=1))
```

### CODE REVIEW

**Comment by lmeyerov** [comment]
Super cool!

I'm wondering:

- What's a good interface? 
     - Typical use: I'm think should probably have something like `motifs_temporal() -> Plottable` so users can `.plot()` right after or continue chaining things
     - .... and have an underlying `motifs_temporal_core() -> DataFrame` closer to current
      - ... where it returns something like a `| node id | motif id` , and i'm unclear on nodes vs edges (`kind="nodes" | "edges" | "both"`, default `"both"` ?) Returning something that matches `g._nodes[[g._node]]` is common to igraph/cugraph/etc: preserves index order, ...
 -  ... what does this actually do (ex: any papers?)  😊 --- cc @silkspace 

Some things to keep in mind:
- Should use mypy on headers and pass lint
- We can add tests as gets more ready
- Stay away from `.iterrows()` and other non-vector ops: for us to run on GPUs, we need written in a vectorized style that we can quickly port to cudf

File: graphistry/motif_utils.py [hunk 1]
```diff
@@ -62,22 +83,22 @@ def motif(
         if self._temporal:
             return motifs
         else:
-            return self.edges(pd.DataFrame(motif_edge_table, columns=[src, dst, "motifID"]))
+            return self.edges(pd.DataFrame(motif_edge_table, columns=[src, dst, motifID]))
 
     @staticmethod
-    def neighbors(graph, v, src, dst):
```
Comment (lmeyerov) [inline graphistry/motif_utils.py]: if this is mixed into the global, should be something like

```
def neighbors(self, v : Union[int, series], mode="ID" | "graph", include_self=False, direction='incoming'|'outgoing'|'both') ->  union[series, plottable]
```

File: graphistry/motif_utils.py [hunk 2]
```diff
@@ -153,14 +174,19 @@ def get_motifs(ETN):
         return '0e'+''.join(e[2:] for e in binary_node_encodings)
 
     @staticmethod
-    def split(x, accumulate, timestamp):
-        times = x[timestamp]
-        pivot = times[0]
+    def _split_edges(edf: pd.DataFrame, accumulate, timestamp):
```
Comment (lmeyerov) [inline graphistry/motif_utils.py:177]: what is this?

**Comment by lmeyerov** [comment]
@tanmoyio can we close?

**Comment by tanmoyio** [comment]
yes

**Pull request closed without merging**